src/h5web/visualizations/shared/utils.ts

```diff
@@ -4,3 +4,16 @@ export const adaptedNumTicks = scaleLinear()
   .domain([300, 900])
   .rangeRound([3, 10])
   .clamp(true);
+
+export function computeSizeFromAspectRatio(
+  availableWidth: number,
+  availableHeight: number,
+  aspectRatio: number
+): { width: number; height: number } {
+  // Determine how to compute canvas size to fit available space while maintaining aspect ratio
+  const shouldAdjustWidth = availableWidth >= availableHeight * aspectRatio;
+
+  return shouldAdjustWidth
+    ? { width: availableHeight * aspectRatio, height: availableHeight }
+    : { width: availableWidth, height: availableWidth * aspectRatio };
+}
```

Comment:
This is the stripped version of the previous hook `useHeatmapStyles`. All the setting of `padding` and `boxSizing` was moved out so that it only computes and returns the width and height.

Reply:
Cool, but why not keep the hook itself with `useMeasure` and the early returns?

Reply:
```const visSize = useHeatmapSize(...)``` -- like we had before but without the padding styles

Reply:
I moved back everything in the function except the padding and the `useMeasure`.